app/ide-desktop/lib/content/src/index.ts

```diff
@@ -119,104 +119,132 @@ function displayDeprecatedVersionDialog() {
 }
 
 // ========================
-// === Main Entry Point ===
+// === Main entry point ===
 // ========================
 
 interface StringConfig {
     [key: string]: StringConfig | string
 }
 
-class Main {
-    async main(inputConfig: StringConfig) {
-        const config = Object.assign(
-            {
-                loader: {
-                    wasmUrl: 'pkg-opt.wasm',
-                    jsUrl: 'pkg.js',
-                    assetsUrl: 'dynamic-assets',
-                },
-            },
-            inputConfig
-        )
-
-        const appInstance = new app.App({
-            config,
-            configOptions: contentConfig.OPTIONS,
-            packageInfo: {
-                version: BUILD_INFO.version,
-                engineVersion: BUILD_INFO.engineVersion,
+// `new app.App` must be run at least once so that the command line arguments are applied to
+// `configOptions.OPTIONS`.
+let currentAppInstance: app.App | null = new app.App({
```

Comment:
whats the point of this global? I though maybe we are using it to support old IDE with authentication but `onAuthenticated` uses `runEnso`. The same if I want to skip auth and new dashboard we also use `runEnso` so both case reinitialise the App which looks like bit of a waste

Reply:
I have removed it and looks like build works fine

Reply:
when i added it back, it was because `contentConfig.OPTIONS` was never changed from defaults, so the check for whether we should start auth/dashboard doesn't succeed no matter what command line arguments are passed

Reply:
ok so re-checked.. When this global is removed it seems that `authentication` and `feature-preview.new-dashboard` flags have default values  even if we pass them in command line.. 

old IDE and debug scenes works so `startup.entry` behaves normally, as other flags

@wdanilo this one is tricky. Is this because of `exportGlobal` in App constructor or some other logic?